pollbot/telegram/callback_handler/creation.py

```diff
@@ -53,20 +56,37 @@ def open_init_anonymization_settings(session, context, poll):
     open_anonymization_settings(context.query.message, poll)
 
 
+@poll_required
+def ask_description(session, context, poll):
+    """Asks user for description of the poll, in case the name was already supplied."""
+    context.user.expected_input = ExpectedInput.description.name
+    keyboard = get_skip_description_keyboard(poll)
+    context.tg_chat.send_message(
+        i18n.t("creation.description", locale=context.user.locale),
+        reply_markup=keyboard,
+    )
+
+
 @poll_required
 def skip_description(session, context, poll):
     """Skip description creation step."""
-    context.user.expected_input = ExpectedInput.options.name
-    session.commit()
-    context.query.message.edit_text(
-        i18n.t("creation.option.first", locale=context.user.locale),
-        reply_markup=get_open_datepicker_keyboard(poll),
-    )
+    if len(poll.options) == 0:
+        context.user.expected_input = ExpectedInput.options.name
+        session.commit()
+        context.query.message.edit_text(
+            i18n.t("creation.option.first", locale=context.user.locale),
+            reply_markup=get_open_datepicker_keyboard(poll),
+        )
+    else:  # options were already prefilled e.g. by native poll
+        # TODO: what if user changed poll type?
+        create_poll(session, poll, context.user, context.tg_chat, context.query.message)
 
 
 @poll_required
 def show_poll_type_keyboard(session, context, poll):
-    """Show to vote type keyboard."""
+    """Show to keyboard to change poll type."""
+
+    # TODO: Review if the poll isn't already available from arguments..? @Nukesor
```

Comment:
Jep, it's definitely already there.
That's probably some really old legacy stuff I forgot to remove while refactoring :sweat_smile:

Reply:
xD 
will also fix the doc